replicaset.go

```diff
@@ -455,21 +448,17 @@ func Set(session mgoSession, members []Member) error {
 	// Compose the added and updated members.
 	var (
 		added   []Member
-		updated []updatedMember
+		updated []Member
 		one     = 1.0
 		votes   = 1
 	)
+	updatedIds := set.NewInts()
 	for _, m := range members {
-		existingAddress, ok := existingIdAddresses[m.Id]
+		existingMember, ok := existingMembers[m.Id]
 		if ok {
-			if existingAddress != m.Address {
-				oldId := m.Id
-				max++
-				m.Id = max
-				updated = append(updated, updatedMember{
-					oldId:  oldId,
-					Member: m,
-				})
+			if existingMember.Address != m.Address {
```

Comment:
Given the complexity of the old code, I think there are cases where you cannot update a member in place. I don't remember exactly when and why, but this feels like something we need to test against a lot of versions of Mongo (2.9 still supports Xenial, so it is 3.2, 3.6, 4.0, 4.4 that we need to test against).

Reply:
I did a smoke test on MAAS with Xenial controllers in HA, and `juju-ha-space` configured. Seems to work fine. Controllers had apt-installed Mongo 3.2.